lib/mini_profiler/config.rb

```diff
@@ -86,6 +86,10 @@ def self.default
 
       attr_reader :assets_url
 
+      def explicit_authorization_mode?
+        [:explicit, :whitelist].include? authorization_mode
```

Comment:
I see no particular reason for keeping this long term ... when you set authorization_mode to :whitelist we should simply print a deprecation and set it to :explicit. then we can remove the deprecation in a few releases. Then we get to avoid recreating this array all the time and the extra method call.

Reply:
Good idea -- I've added 4b1f3aae9c1c3d3f59b58c1d3bd9451ef043cd7b which takes that approach using `Warning` to emit the deprecation message. Not sure if the latest of this MR will be satisfactory or if we'd prefer something like `!!logger ? logger.warn(msg) : warn(msg)`.

Thanks for the suggestion. Happy to make further adjustments as needed!